docs/source/task_guides/int8-asr.mdx

```diff
@@ -0,0 +1,376 @@
+
+```py
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+import numpy as np
+import gc
+
+eval_dataloader = DataLoader(common_voice["test"], batch_size=8, collate_fn=data_collator)
+
+model.eval()
+for step, batch in enumerate(tqdm(eval_dataloader)):
+    with torch.cuda.amp.autocast():
+        with torch.no_grad():
+            generated_tokens = (
+                model.generate(
+                    input_features=batch["input_features"].to("cuda"),
+                    decoder_input_ids=batch["labels"][:, :4].to("cuda"),
+                    max_new_tokens=255,
+                )
+                .cpu()
+                .numpy()
+            )
+            labels = batch["labels"].cpu().numpy()
+            labels = np.where(labels != -100, labels, tokenizer.pad_token_id)
+            decoded_preds = tokenizer.batch_decode(generated_tokens, skip_special_tokens=True)
+            decoded_labels = tokenizer.batch_decode(labels, skip_special_tokens=True)
+            metric.add_batch(
+                predictions=decoded_preds,
+                references=decoded_labels,
+            )
+    del generated_tokens, labels, batch
+    gc.collect()
+wer = 100 * metric.compute()
+print(f"{wer=}")
+```
+
+## Share model
+
+Once you're happy with your results, you can upload your model to the Hub with the [`~transformers.PreTrainedModel.push_to_hub`] method:
+
+```py
+model.push_to_hub("your-name/int8-whisper-large-v2-asr")
+```
+
+## Inference
+
+Let's test the model out now!
+
+Instantiate the model configuration from [`PeftConfig`], and from here, you can use the configuration to load the base and [`PeftModel`], tokenizer, processor, and feature extractor. Remember to define the `language` and `task` in the tokenizer, processor, and `forced_decoder_ids`:
+
+```py
+from peft import PeftModel, PeftConfig
+
+peft_model_id = "smangrul/openai-whisper-large-v2-LORA-colab"
+language = "Marathi"
+task = "transcribe"
+peft_config = PeftConfig.from_pretrained(peft_model_id)
+model = WhisperForConditionalGeneration.from_pretrained(
+    peft_config.base_model_name_or_path, load_in_8bit=True, device_map="auto"
+)
+model = PeftModel.from_pretrained(model, peft_model_id)
+tokenizer = WhisperTokenizer.from_pretrained(peft_config.base_model_name_or_path, language=language, task=task)
+processor = WhisperProcessor.from_pretrained(peft_config.base_model_name_or_path, language=language, task=task)
+feature_extractor = processor.feature_extractor
+forced_decoder_ids = processor.get_decoder_prompt_ids(language=language, task=task)
+```
+
+Load an audio sample (download the audio sample to listen to it) to transcribe, and the [`~transformers.AutomaticSpeechRecognitionPipeline`]:
+
+```py
+from transformers import AutomaticSpeechRecognitionPipeline
+
+audio = "https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/1.flac"
```

Comment:
Can we maybe use a Marathi audio clip, as the current one is English and the model somehow managed to do a bit of translation as well of transliteration (interesting behaviour) but falls short of coherence?